File: 0019-remove-nth-node-from-end-of-list/0019-remove-nth-node-from-end-of-list.js
const removeNthFromEnd = (head, n) => {
  const sentinel = new ListNode(0, head);

  let left = sentinel;
  let right = sentinel;

  for (let i = 0; i < n + 1; i++) {
    right = right.next;
  }

  while (right !== null) {
    left = left.next;
    right = right.next;
  }

  left.next = left.next.next;

  return sentinel.next;
};

File: 0019-remove-nth-node-from-end-of-list/0019-remove-nth-node-from-end-of-list.py
class Solution:
    def removeNthFromEnd(self, head: Optional[ListNode], n: int) -> Optional[ListNode]:
        sentinel = ListNode(0, head)

        left = sentinel
        right = sentinel

        for i in range(n + 1):
            right = right.next
        
        while (right):
            left = left.next
            right = right.next
        
        left.next = left.next.next
        
        return sentinel.next
        